hydra/_internal/core_plugins/basic_sweeper.py

```diff
@@ -152,8 +154,26 @@ def sweep(self, arguments: List[str]) -> Any:
                 f"Validated configs of {len(batch)} jobs in {elapsed:0.2f} seconds, {len(batch)/elapsed:.2f} / second)"
             )
             results = self.launcher.launch(batch, initial_job_idx=initial_job_idx)
+
+            idx = initial_job_idx
+
+            for r in results:
+                try:
+                    _ = r.return_value
+                except Exception as e:
+                    job_failed.append(idx)
+                    log.error(f"Job #{idx} failed.")
+                    if first_exception is None:
+                        first_exception = e
+                idx += 1
+
             initial_job_idx += len(batch)
             returns.append(results)
+
+        if first_exception is not None:
+            raise first_exception from FailedMultirunException(
+                f"Multirun failed, failed job: {','.join(['#' + str(j) for j in job_failed])}"
+            )
```

Comment:
I think that in order to keep the basic launcher basic, we should just let the first exception fly and terminate the loop.

just access the result to trigger the exception.